ddtrace/settings/config.py

```diff
@@ -141,6 +159,229 @@ def get_error_ranges(error_range_str):
     return error_ranges  # type: ignore[return-value]
 
 
+_ConfigResult = TypedDict("_ConfigResult", {"value": Any, "raw": Any, "source": Any})
+
+
+class _ConfigSource(object):
+    """A configuration source that can be queried for a value."""
+
+    def get(self):
+        # type: () -> _ConfigResult
+        raise NotImplementedError
+
+
+class _ConfigSourceEnv(_ConfigSource):
+    """A configuration source that reads from an environment variable."""
+
+    def __init__(
+        self,
+        name,  # type: str
+        factory=lambda x: x,  # type: Callable[[str], Any]
+        examples=None,  # type: Optional[List[str]]
+    ):
+        # type: (...) -> None
+        self.key = name
+        self.factory = factory
+        self.examples = examples or []
+
+    def get(self):
+        # type: () -> _ConfigResult
+        raw = os.environ.get(self.key)
```

Comment:
would it be hard to achieve the same goal of reading from the environment with envier here? This refactor PR might be a good first step towards creating a global config and aligning the style with other product configurations.